### PR TITLE
[test-server] Add test server to run conformance tests.

### PR DESCRIPTION
**Pull request by granaghan**
Adds a utility the creates sockets to conform to the TPM2 simulator spec and forwards them to a physical device.

### CODE REVIEW

File: sw/host/opentitanlib/src/tpm/access.rs [hunk 1]
```diff
@@ -0,0 +1,29 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+pub struct TpmAccess(u8);
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/tpm/access.rs:5]: Would this struct be better modeled with the `bitflags!` macro?

File: sw/host/opentitanlib/src/tpm/status.rs [hunk 2]
```diff
@@ -0,0 +1,48 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::fmt;
+
+pub struct TpmStatus(u32);
```
Comment (cfrantz) [inline sw/host/opentitanlib/src/tpm/status.rs:7]: Would this struct be better modeled with the `bitflags!` macro?

File: sw/host/tpm2_test_server/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+        default_value = "9883",
+        help = "TCP port for incoming connections."
+    )]
+    tpm_port: u16,
+}
+
+const CMD_TOKEN: Token = Token(0);
+const PLATFORM_TOKEN: Token = Token(1);
+
+pub fn main() -> std::io::Result<()> {
```
Comment (cfrantz) [inline sw/host/tpm2_test_server/src/main.rs:56]: What program will be the client to `tpm2_test_server`?

Comment (granaghan) [reply sw/host/tpm2_test_server/src/main.rs:56]: Either [Tpm2Tester](https://github.com/TrustedComputingGroup/Compliance_PC-TPM-Internal), a sample test suite, or [Comliance_PC-TPM-Internal](https://github.com/TrustedComputingGroup/Compliance_PC-TPM-Internal), a test suite from TCG available to members.

File: sw/host/opentitanlib/src/tpm/access.rs (hunk 1 above)
Comment (granaghan) [reply sw/host/opentitanlib/src/tpm/access.rs:5]: Yes, will update.

File: sw/host/opentitanlib/src/tpm/status.rs (hunk 2 above)
Comment (granaghan) [reply sw/host/opentitanlib/src/tpm/status.rs:7]: burst_count is a 16 bit field, so this one may not be better as a `bitflags!`.

**Review by cfrantz — APPROVED** [review]

**Pull request merged**